models/comments.models.js

```diff
@@ -0,0 +1,16 @@
+const db = require("../db/connection.js")
+
+exports.selectCommentsByArticleID = (article_id) => {
+    return db
+    .query(`SELECT * FROM comments WHERE article_id = $1
+    ORDER BY created_at DESC`, [article_id])
+    .then((comments) => {
+        if(comments.rows.length === 0) {
```

Comment:
This logic is great for non-existent articles! However it will also wrongly trigger the Promise.reject for existing articles with no attached comments (both return []). 

We may need to check the article exists _before_ executing any logic on returning empty rows. There are many ways to do this, see the notes! 

https://notes.northcoders.com/courses/js-back-end/advanced-error-handling#sending-the-correct-response

You could potentially use a previous model function that gets an article by Id, if this doesn't reject, you _know_ the article is valid, and can safely return rows regardless of length 👍